### PR TITLE
refactor(table-card): remove styled components

### LINKED ISSUE #3315: [TableCard] Convert styled-components to scss
We need to remove styles that are done using styled components and remove the package as a dependency. It may make sense to do a local visual snapshot with Cypress before removal to tell if we are introducing any regressions. Concentrate on deprecated components and stories last.

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #3315 

**Summary**

- removes styled components from TableCard

**Change List (commits, features, bugs, etc)**

- removes styled components from TableCard
- removes jest test related to styled components in expanded state and moves it to cypress
- updates snapshots

**Acceptance Test (how to verify the PR)**

- tests pass 

**Regression Test (how to make sure this PR doesn't break old functionality)**

- no visual changes to [TableCard stories](https://deploy-preview-3325--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-tablecard--with-multiple-actions)

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/Dashboard/DashboardGrid.test.e2e.jsx [hunk 1]
```diff
@@ -1162,6 +1162,7 @@ DashboardAllCardsAsResizable.propTypes = {
 describe('DashboardGrid', () => {
   beforeEach(() => {
     MockDate.set(1537538254000);
+    cy.viewport(1680, 900);
```
Comment (bjornalm) [inline packages/react/src/components/Dashboard/DashboardGrid.test.e2e.jsx:1165]: Why are we setting the viewport to the same thing before and after all?

File: packages/react/src/components/TableCard/_table-card.scss [hunk 2]
```diff
@@ -52,3 +52,105 @@
...
+    tr {
+      height: $spacing-07;
+    }
+  }
+
+  .#{$prefix}--data-table tbody tr {
+    height: $spacing-08;
+  }
+  .#{$prefix}--data-table-content + .#{$prefix}--pagination {
+    border: 1px solid #dfe3e6;
```
Comment (bjornalm) [inline packages/react/src/components/TableCard/_table-card.scss:94]: Special color or miss?

**Review by bjornalm — APPROVED** [review]
Looks good, just a question and possibly the hard coded color.

File: packages/react/src/components/TableCard/_table-card.scss (hunk 2 above)
Comment (kevinsperrine) [reply packages/react/src/components/TableCard/_table-card.scss:94]: Just missed it. I'll find it's proper token. Thanks!

File: packages/react/src/components/Dashboard/DashboardGrid.test.e2e.jsx (hunk 1 above)
Comment (kevinsperrine) [reply packages/react/src/components/Dashboard/DashboardGrid.test.e2e.jsx:1165]: I've realized cypress loses this value often. That's why in some of the automated tests items will get pushed off screen until you resize, so I've started adding it to tests as I come across it to keep the viewport rendering correctly. It's probably overkill to have it in both before/after here though.

**Review by blechner — APPROVED** [review]

File: packages/react/src/components/TableCard/_table-card.scss [hunk 3]
```diff
@@ -52,3 +52,105 @@
...
+    height: 100%;
+    td {
+      white-space: nowrap;
+    }
+  }
+  .#{$prefix}--data-table thead tr:nth-child(2) {
+    height: $spacing-10;
+
+    th {
+      padding-top: 5px;
```
Comment (davidicus) [inline packages/react/src/components/TableCard/_table-card.scss:110]: did these not scream at you in our linter for using a pixel value?

**Review by davidicus — APPROVED** [review]
This looks good. We may want to use the rem function to convert our pixel values. I am  not sure why the linter did not bark for this.

File: packages/react/src/components/TableCard/_table-card.scss (hunk 3 above)
Comment (kevinsperrine) [reply packages/react/src/components/TableCard/_table-card.scss]: Nope, I just tried running it and scanning that file directly, and it still passes. Not sure what's going on there. I wrapped all the px with rem them.

**Pull request merged**